src/Data/Group.hs

```diff
@@ -101,3 +123,62 @@ generated :: Cyclic a => [a]
 generated =
   iterate (mappend generator) mempty
 
+instance Cyclic () where
+  generator = ()
+
+#if MIN_VERSION_base(4,7,0)
+-- | Trivial group, Functor style.
+instance Group (Proxy x) where
+  invert _ = Proxy
+  _ ~~ _ = Proxy
```

Comment:
I think an explicit definition of `pow` is worth it here and for other instances that are isomorphic to `()`

Reply:
Done. I think `Proxy` is the only such type?